packages/components/src/box-control/axial-input-controls.tsx

```diff
@@ -128,26 +137,62 @@ export default function AxialInputControls( {
 					side === 'vertical'
 						? selectedUnits.top
 						: selectedUnits.left;
+
+				const inputId = [ generatedId, side ].join( '-' );
+
 				return (
-					<UnitControl
-						{ ...props }
-						isFirst={ first === side }
-						isLast={ last === side }
-						isOnly={ only === side }
-						value={ [
-							parsedQuantity,
-							selectedUnit ?? parsedUnit,
-						].join( '' ) }
-						onChange={ createHandleOnChange( side ) }
-						onUnitChange={ createHandleOnUnitChange( side ) }
-						onFocus={ createHandleOnFocus( side ) }
-						onHoverOn={ createHandleOnHoverOn( side ) }
-						onHoverOff={ createHandleOnHoverOff( side ) }
-						label={ LABELS[ side ] }
-						key={ side }
-					/>
+					<InputWrapper key={ `box-control-${ side }` }>
```

Comment:
Nit: This is a React `key` and not an HTML `id`, so we only have to worry about uniqueness among the items of this particular array (which is `filteredSides`).

```suggestion
					<InputWrapper key={ side }>
```

Reply:
A very helpful nit — thank you!